### PR TITLE
*: add initial read benchmark for etcd v3

### PR DESCRIPTION
**Pull request by yichengq**
It includes the initial read benchmark for etcd v3.

This is the first step to give some rough thoughts. I haven't digged
deeper to answer some questions, including why its performance is not
better than HTTP + json, why one put will cause performance downgrade.

for #3290

### CODE REVIEW

File: hack/v3benchmark/main.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	"sync"
+	"time"
+
+	"github.com/coreos/etcd/Godeps/_workspace/src/golang.org/x/net/context"
+	"github.com/coreos/etcd/Godeps/_workspace/src/google.golang.org/grpc"
+	"github.com/coreos/etcd/etcdserver/etcdserverpb"
+	"github.com/rakyll/pb"
+)
+
+func main() {
```
Comment (xiang90) [inline hack/v3benchmark/main.go]: can we move this to /tools?

Comment (xiang90) [inline hack/v3benchmark/main.go]: Also I would suggest to check https://github.com/grpc/grpc-go/blob/master/benchmark/benchmark.go

**Comment by yichengq** [comment]
Moved.

I checked the benchmark written by gRPC, and finds that it uses one connection for all requests in concurrent way, while I use multiple connections, each of which sends one request at one time. I have added a TODO in code to support that.

Having played with there two methods, I find using one connection could perform 10% better when having 256 clients, and performs nearly the same in other situations. There is still gap between its performance and expectation. The benchmark today cannot match the grpc internal benchmark, which will be researched soon. 

For etcd, i would expect the use case is O(100-1000) connections with O(1) concurrency in each connection.

**Comment by xiang90** [comment]
We can merge as it is today. I just briefly look through the code and did not give it a careful read.

File: tools/v3benchmark/main.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
+package main
```
Comment (xiang90) [inline tools/v3benchmark/main.go]: add coreos licensee

**Comment by yichengq** [comment]
license added.

It performs the same way as boom does, and nearly the same way as grpc-benchmark does except the problem that how many concurrency on one connection.

**Comment by xiang90** [comment]
Let merge this as it is. This would be a good start.

**Comment by yichengq** [comment]
:ship:

**Pull request merged**